site/src/layouts/DetailComponent/DetailComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC, useCallback, useMemo } from "react";
+import { useRouter } from "next/navigation";
+import {
+  TableOfContents,
+  Sidebar,
+} from "@jpmorganchase/mosaic-site-components";
+import { useRoute, useStore, SiteState } from "@jpmorganchase/mosaic-store";
+import { TabPanel, Tabs } from "@salt-ds/lab";
+import { LayoutProps } from "../types/index";
+import { DetailBase } from "../DetailBase";
+
+const tabs = [
+  { id: 0, name: "examples", label: "Examples" },
+  { id: 1, name: "usage", label: "How to use" },
+  { id: 2, name: "accessibility", label: "Accessibility" },
+];
+
+type CustomSiteState = SiteState & { data?: Record<string, any> };
+
+export const DetailComponent: FC<LayoutProps> = ({ children }) => {
+  const { push } = useRouter();
+  const { route } = useRoute();
+
+  const newRoute = route?.substring(0, route.lastIndexOf("/") + 1);
+
+  const useData = useStore((state: CustomSiteState) => state.data);
+
+  const { description } = useData || {};
+
+  const updateRouteWhenTabChanges = useCallback((index: number) => {
+    const currentTab = tabs.find(({ id }) => index === id);
+
+    currentTab
+      ? push(`${newRoute}${currentTab.name}`)
+      : push(`${newRoute}${tabs[0].name}`);
+  }, []);
+
+  const updateTabWhenRouteChanges = useMemo(() => {
```

Comment:
This useMemo is not needed and the variable name should be updated so it doesn't seem like this is a function